### ISSUE TITLE
Validation not compliant when using bnodes?

### ISSUE THREAD

**Issue by vcharpenay** [TopQuadrant/shacl-js#21]
I tested on the playground with the following shape:

```turtle
@prefix sh: <http://www.w3.org/ns/shacl#> .
@prefix ex: <http://example.org/> .

ex:ClassShape a sh:NodeShape ;
  sh:targetClass ex:C ;
  sh:property [
    sh:path ex:p ;
    sh:maxCount 1
  ] .
```

As expected, the following instance has no violation: 

```turtle
@prefix ex: <http://example.org/> .

<tag:b0> a ex:C ; ex:p 0 . 
<tag:b1> a ex:C ; ex:p 1 .
```

However, if I use blank nodes as follows, I get a max count violation:

```turtle
@prefix ex: <http://example.org/> .

_:b0 a ex:C ; ex:p 0 . 
_:b1 a ex:C ; ex:p 1 .
```

This should make no difference according  to the standard, right?

**Comment by HolgerKnublauch** [comment]
Yes this looks incorrect. I do believe the SHACL engine itself is correct but there might be an issue with how matching of bnodes is implemented in that rdflib.js version. The query is here

https://github.com/TopQuadrant/shacl-js/blob/master/shared/dash.js#L218

and $this is the focus node (i.e. the blank nodes). I don't know how rdflib treats equality of bnodes during the PathQuery execution. This is of course a requirement for the upper code (in dash.js) to work correctly. Does anyone see something wrong?

**Comment by michaelcpuckett** [comment]
Both ways pass validation when using this fork:

https://github.com/zazuko/shacl-js
